src/main/kotlin/com/hypto/iam/server/security/Authorization.kt

```diff
@@ -112,7 +109,7 @@ class Authorization(config: Configuration) : KoinComponent {
                         val actionHrn = ActionHrn(resourceHrn.organization, null, resourceHrn.resource, it)
                         PolicyRequest(principalHrn.toString(), resourceHrn.toString(), actionHrn.toString())
                     }.toList()
-                if (!policyValidator.validate(policies.stream(), policyRequests)) {
+                if (!PolicyValidator().validate(policies.stream(), policyRequests)) {
```

Comment:
creating an instance of PolicyValidator reads casbin_conf file from disk, parses it and applies the policy against it before validation the request. Reading a file for each request will deteriorate performance by a lot.